src/library/scala/collection/Stepper.scala

```diff
@@ -69,15 +69,15 @@ trait Stepper[@specialized(Double, Int, Long) A] {
     * a [[Spliterator.OfInt]] (which is a `Spliterator[Integer]`) in the subclass [[IntStepper]]
     * (which is a `Stepper[Int]`).
     */
-  def spliterator: Spliterator[_]
+  def spliterator[B >: A]: Spliterator[_]
```

Comment:
What's the tparam for?

Reply:
In the `AnyStepper[+A]` subclass, having `def spliterator: Spliterator[A]` doesn't work because `Spliterator` is invariant..

Reply:
In AnyStepper, couldn't you write:
```scala
def spliterator: Spliterator[_ >: A]
```

Reply:
Hmm, seems not...
```scala
  class AnyStepperSpliterator[A](s: AnyStepper[A]) extends Spliterator[A] {
    def trySplit(): Spliterator[A] = {
      val sp = s.trySplit()
      if (sp == null) null else (sp.spliterator: Spliterator[A])
    }
```
```
Error:(202, 37) type mismatch;
 found   : java.util.Spliterator[_$3] where type _$3 >: A
 required: java.util.Spliterator[A]
Note: _$3 >: A, but Java-defined trait Spliterator is invariant in type T.
You may wish to investigate a wildcard type such as `_ >: A`. (SLS 3.2.10)
      if (sp == null) null else (sp.spliterator: Spliterator[A])
```

Reply:
Oh, and also

```scala
trait AnyStepper[+A] extends Stepper[A] {
  def spliterator: Spliterator[_ >: A] = new AnyStepper.AnyStepperSpliterator(this)
```

```
Error:(188, 7) covariant type A occurs in contravariant position in type => java.util.Spliterator[_ >: A] of method spliterator
  def spliterator: Spliterator[_ >: A] = new AnyStepper.AnyStepperSpliterator(this)
```

Reply:
Does Spliterator actually behaves covariantly ? If so an @uncheckedVariance annotation might be warranted.

Reply:
Ticket for the existential type inference bug: https://github.com/scala/bug/issues/10920

Reply:
> Does Spliterator actually behaves covariantly ? If so an @uncheckedVariance annotation might be warranted.

Hmm, that worries me more than the type parametr on `def spliterator` :-)

```java
public interface Spliterator<T> {
    default Comparator<? super T> getComparator() { ... }
}
```

If I translate that to Scala

```scala
trait SSpliterator[+T] {
  def getComparator: Comparator[_ >: T]
}
```
```
Error:(22, 7) covariant type T occurs in contravariant position in type => java.util.Comparator[_ >: T] of method getComparator
  def getComparator: Comparator[_ >: T]
```

Also, everyone overriding `def spliterator` would have to repeat the `@uncheckedVariance`, which is a not a good option I think.